### PR TITLE
fix EventuallyPredicate, add a producerless version, and inform user …

### PR DESCRIPTION
**Pull request by jschneidereit**
…about predicate failures for #2044

### CODE REVIEW

**Comment by xeruf** [comment]
yeah, might as well call it `until` again ^^

Either way, you need to rework or remove this, as it is incorrect:
```kt
suspend fun until(duration: Duration, interval: Interval = 1.seconds.fixed(), f: suspend () -> Boolean) =
   eventually(duration, interval, f = f)
```
It will return immediately after the first try if that doesn't throw an exception (which until lambdas are not supposed to do) regardless of the returned boolean value.

**Comment by sksamuel** [comment]
> It will return immediately after the first try if that doesn't throw an exception (which until lambdas are not supposed to do) regardless of the returned boolean value.

That's how eventually has always worked - it looked for exceptions.

Now you can additionally evaluate the predicate, combining the power of both eventually and until.

File: kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/timing/eventually.kt [hunk 1]
```diff
@@ -57,6 +57,18 @@ suspend fun <T> eventually(duration: Duration, poll: Duration, f: SuspendingProd
 suspend fun <T> eventually(duration: Duration, exceptionClass: KClass<out Throwable>, f: SuspendingProducer<T>): T =
    eventually(EventuallyConfig(duration = duration, exceptionClass = exceptionClass), f = f)
 
+/**
+ * Runs a predicate until the predicate returns true as long as the specified duration hasn't passed
+ */
+suspend fun eventuallyPredicate(
```
Comment (sksamuel) [inline kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/timing/eventually.kt]: Can we make this work without requiring the name overload ?

**Comment by xeruf** [comment]
> That's how eventually has always worked - it looked for exceptions.

Yes, but `until` never worked like that, so this alias was bogus.

**Comment by sksamuel** [comment]
4.4.1 has been released with until restored and documented

**Pull request merged**